doc/docusaurus/docs/delve-deeper/languages.md

```diff
@@ -0,0 +1,100 @@
+When using an embedded DSL, you are essentially writing programs that create and manage ASTs, rather than straightforward code.
+
+Take, for instance, a program that accepts two integers as input, and checks if the first is less than the second.
+Normally, you would write a function of type `Integer -> Integer -> Bool`, which takes two integers and returns a boolean.
+However, when working with an eDSL, your program might have a type like `AST Integer -> AST Integer -> AST Bool`, which takes two ASTs that evaluate to integers, combines them, and yields a larger AST that evaluates to a boolean.
+Like standalone DSLs, this also introduces additional learning curves and cognitive load, though for a different reason.
+
+Another disadvantage of eDSLs is that it is harder, compared to the other two approaches, to produce readable target code or accurate source mappings for debuggers.
+This stems from the nature of eDSLs, which are libraries that construct and manipulate ASTs.
+Since they do not have direct access to the host language's ASTs, it can be challenging to retrieve information related to the source code, such as variable names, module names and code locations.
+
+There's another category of eDSLs, known as "shallow embedding", which, unlikely the "deep embedding" described earlier, does not construct intermediate ASTs.
+Instead, shallow embedding involves using overloaded functions.
+For example, a DSL designed as a shallow embedding for working with databases might include operations such as `createTable`, `getItem`, and `putItem`.
+These functions are overloaded, allowing them to work with various database implementations, including mock databases for testing purposes.
+Such overloaded functions are typically defined using typeclasses in functional languages, or interfaces/traits in object-oriented languages.
+
+While it is valid to call shallow embeddings _languages_, it is a bit of a stretch.
+Overloaded functions are widespread in everyday programming, and are not usually regarded as languages due to the absence of ASTs.
+All existing eDSLs targeting UPLC are examples of deep embeddings.
+
+## Subsets of Existing Languages
+
+Similar to eDSLs, this approach can be particularly appealing if your team or project is already using the host language.
+It allows for even greater reuse of existing functions, types and idioms from the hosting language, compared to eDSLs.
+For instance, a program that tests whether one integer is less than another can retain the type `Integer -> Integer -> Bool`, and can even reuse the `<` operator in the hosting language's standard library[^1].
+
+This is achieved by leveraging the host language's compiler components, such as lexer, parser, type checker and optimization passes, as well as the hosting language's intermediate ASTs.
+By reusing the host language's ASTs, programs maintain simple and regular types without the need for custom AST construction, which is often necessary in eDSLs.
```

Comment:
I think something's missing in this section.  For example OpShin is a subset of Python, but you don't just compile it using the usual Python tools and deploy the result on the chain.  I don't know exactly what happens in the case of OpShin (and the other languages mentioned here), but something else must be required to get UPLC out of the Python tools, like a new backend for the Python compiler or a translator from the output of the Python compiler (bytecode?) to UPLC.  If your language was a subset of Python, say, you could even write a totally separate tool that compiles it to UPLC but you could use the standard Python tools to develop and debug programs.   The point is that (as this paragraph points out) being able to re-use some of the existing infrastructure saves a lot of effort; you still need to add something though.

Reply:
I rephrased the first sentence of this paragraph to: "This is achieved by leveraging the host language's compiler frontend, which might include lexer, parser, type checker, AST and optimization passes, while developing a custom backend for the new language."